src/probability/integration.lean

```diff
@@ -118,4 +120,132 @@ lintegral_mul_eq_lintegral_mul_lintegral_of_independent_measurable_space
   (measurable.of_comap_le le_rfl) (measurable.of_comap_le le_rfl)
 
+/-- This proves that if `f` and `g` are independent and almost everywhere measurable,
+   then `E[f * g] = E[f] * E[g]` (slightly generalizing
+   `lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun`). -/
+lemma lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun' [measurable_space α] {μ : measure α}
+  {f g : α → ℝ≥0∞} (h_meas_f : ae_measurable f μ) (h_meas_g : ae_measurable g μ)
+  (h_indep_fun : indep_fun f g μ) :
+  ∫⁻ a, (f * g) a ∂μ = ∫⁻ a, f a ∂μ * ∫⁻ a, g a ∂μ :=
+begin
+  rcases h_meas_f with ⟨f',f'_meas,f'_ae⟩,
+  rcases h_meas_g with ⟨g',g'_meas,g'_ae⟩,
+  have fg_ae : f * g =ᵐ[μ] f' * g' := f'_ae.mul g'_ae,
+  rw [lintegral_congr_ae f'_ae, lintegral_congr_ae g'_ae, lintegral_congr_ae fg_ae],
+  apply lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun f'_meas g'_meas,
+  exact h_indep_fun.ae_eq f'_ae g'_ae
+end
+
+/-- This shows that the product of two independent, integrable, real_valued random variables
+  is itself integrable. -/
+lemma indep_fun.integrable_mul [measurable_space α] {μ : measure α}
+  {β : Type*} [measurable_space β] [normed_division_ring β] [borel_space β]
+  {X Y : α → β} (hXY : indep_fun X Y μ) (hX : integrable X μ) (hY : integrable Y μ) :
+  integrable (X * Y) μ :=
+begin
+  let nX : α → ennreal := λ a, ∥X a∥₊,
+  let nY : α → ennreal := λ a, ∥Y a∥₊,
+
+  have hXY' : indep_fun (λ a, ∥X a∥₊) (λ a, ∥Y a∥₊) μ :=
+    hXY.comp measurable_nnnorm measurable_nnnorm,
+  have hXY'' : indep_fun nX nY μ :=
+    hXY'.comp measurable_coe_nnreal_ennreal measurable_coe_nnreal_ennreal,
+
+  have hnX : ae_measurable nX μ := hX.1.ae_measurable.nnnorm.coe_nnreal_ennreal,
+  have hnY : ae_measurable nY μ := hY.1.ae_measurable.nnnorm.coe_nnreal_ennreal,
+
+  have hmul : ∫⁻ a, nX a * nY a ∂μ = ∫⁻ a, nX a ∂μ * ∫⁻ a, nY a ∂μ :=
+    by convert lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun' hnX hnY hXY'',
+
+  refine ⟨hX.1.mul hY.1, _⟩,
+  simp_rw [has_finite_integral, pi.mul_apply, nnnorm_mul, ennreal.coe_mul, hmul],
+  exact ennreal.mul_lt_top_iff.mpr (or.inl ⟨hX.2, hY.2⟩)
+end
+
+/-- This shows that the (Bochner) integral of the product of two independent, nonnegative random
+  variables is the product of their integrals. The proof is just plumbing around
+  `lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun'`. -/
+lemma indep_fun.integral_mul_of_nonneg [measurable_space α] {μ : measure α} {X Y : α → ℝ}
+  (hXY : indep_fun X Y μ) (hXp : 0 ≤ X) (hYp : 0 ≤ Y)
+  (hXm : ae_measurable X μ) (hYm : ae_measurable Y μ) :
+  integral μ (X * Y) = integral μ X * integral μ Y :=
+begin
+  have h1 : ae_measurable (λ a, ennreal.of_real (X a)) μ :=
+    ennreal.measurable_of_real.comp_ae_measurable hXm,
+  have h2 : ae_measurable (λ a, ennreal.of_real (Y a)) μ :=
+    ennreal.measurable_of_real.comp_ae_measurable hYm,
+  have h3 : ae_measurable (X * Y) μ := hXm.mul hYm,
+
+  have h4 : 0 ≤ᵐ[μ] X := ae_of_all _ hXp,
+  have h5 : 0 ≤ᵐ[μ] Y := ae_of_all _ hYp,
```

Comment:
these two are short and used only once: you could use the proofs directly instead of using `have` for them.

Reply:
Done.